---
title: "Comparison mitochrodia gene expression between foveal RGC cells and peripheial RGC cells in human and macauqe data respectively"
output:
  html_document:
    toc: true

---





# Human data, data was downloaded from
https://singlecell.broadinstitute.org/single_cell/study/SCP839/cell-atlas-of-the-human-fovea-and-peripheral-retina#study-download
Human_retina_combined_all_expression_matrix.csv
HumanRGC_clusterfile.txt 


```{r}
#pak1 = c("rmarkdown", "readr","Seurat","cowplot","patchword","ggplot2","tidyverse","tidyverse",
#"data.table", "fgsea", "clusterProfiler", "enrichiplot", "export", "rmarkdown")

#if (!require(pak1)) {
#install.packages(pak1)
#}


library(rmarkdown)
library(readr)
library(Seurat)
library(cowplot)
library(patchwork) 
library(ggplot2)
library(tidyverse)
library(data.table)


library(edgeR)
library(DESeq2)
library(fgsea)
library(clusterProfiler)
library(enrichplot)
library(export)


RGC <- fread(file="human_fRGC_pRGC/SCP839/expression/Human_retina_combined_all_expression_matrix.csv")  %>% as.data.frame()

#RGC  = read_csv("human_fRGC_pRGC/SCP839/expression/Human_retina_combined_all_expression_matrix.csv")

row.names(RGC) <- RGC$V1


#import cluster file
clustered_RGC <-
  read.delim(
    "human_fRGC_pRGC/SCP839/cluster/HumanRGC_clusterfile.txt",
    header = F,
    sep = "\t",
    dec = "."
  )

clustered_RGC = clustered_RGC[-1,]
clustered_RGC = clustered_RGC[-1,]

#assign RGC cluster info to expression file
RGC = RGC %>% select(clustered_RGC$V1)


```



```{r}


#build the suerat object
Human_Object = CreateSeuratObject(
  RGC,
  project = "SeuratProject",
  assay = "RNA",
  min.cells = 50,
  min.features = 200
)



```



```{r}
# rename 13 fRFC clusters to a foveal group
n_f = 'foveal'
Human_Object <- RenameIdents(
  Human_Object,
  'H11FoveaS1' = n_f,
  'H2Fovea1S1' = n_f,
  'H2Fovea2S1' = n_f,
  'H3FoveaS1' = n_f,
  'H3FoveaS2' = n_f,
  'H3FoveaS3' = n_f,
  'H4FoveaS1' = n_f,
  'H5FoveaS1' = n_f,
  'H5FoveaS2' = n_f,
  'H5FoveaS3' = n_f,
  'H5FoveaS4' = n_f,
  'H5FoveaS5' = n_f,
  'H9FoveaS1' = n_f
)
# rename 6 pRGC clusters to a peripheial group
n_p = 'peripheral'
Human_Object <- RenameIdents(
  Human_Object,
  'H1CD73dpS1' = n_p,
  'H1CD90S1' = n_p,
  'H3CD73dpS1' = n_p,
  'H3CD73dpS2' = n_p,
  'H3CD90S1' = n_p,
  'H3CD90S2' = n_p
)


```



```{r}
#gene differential analysis between foveal and peripheral cell 

MAST_Human_20 = FindMarkers(
  Human_Object ,
  ident.1 = "foveal",
  ident.2 = "peripheral",
  test.use = "MAST" ,
  logfc.threshold = 0.25,
  assay = "RNA",
  verbose = FALSE
)

```


## tabulate expression result of MRP gene
```{r}
# tabulate expression result of MRP gene
MAST_Human_20$Gene = rownames(MAST_Human_20)
mrp_List_20 =  MAST_Human_20  %>%  filter( grepl('MRP', Gene))


mrp_List_20
```



## Plot, Mitochondrial Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)

```{r}
library(EnhancedVolcano)
mrg = EnhancedVolcano(mrp_List_20,
                      lab = rownames(mrp_List_20) ,
                      x = 'avg_log2FC',
                      y = 'p_val_adj',
                      xlim = c(-1, +1),
                      ylim = c(-40,300),
                      title = "Mitochondrial Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)",
                      FCcutoff = 0.24,
                      labSize = 4,
                      drawConnectors = TRUE,
                      arrowheads = TRUE,
                      colConnectors = 'black',
                      legendPosition = 'top',
                      legendLabSize = 10,
                      legendIconSize = 5
)

mrg
```



## Plot, Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)

```{r}

rp_List1 =  MAST_Human_20  %>%  filter( grepl(paste(c('^RPL','^RPS'), collapse = "|") ,Gene
)
)

rp = EnhancedVolcano(rp_List1,
                     lab = rownames(rp_List1) ,
                     x = 'avg_log2FC',
                     y = 'p_val_adj',
                     xlim = c(-2,2),
                     ylim = c(0,350),
                     title = "Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)",
                     FCcutoff = 0.24,
                     labSize = 4,
                     drawConnectors = TRUE,
                     arrowheads = TRUE,
                     colConnectors = 'black',
                     legendPosition = 'top',
                     legendLabSize = 10,
                     legendIconSize = 5
)
rp


```



## GSEA


```{r}

library(edgeR)
library(DESeq2)
library(fgsea)
library(clusterProfiler)
library(enrichplot)
library(ggplot2)
library(ggpubr)
library(export)



# GSEA USING ALL Genome June 14 ,
# GEST hallmark


# get the genome wide fold change table and sort
human_fold = FoldChange(Human_Object,
                        ident.1 = "foveal",
                        ident.2 = "peripheral")

human_fold <- human_fold[order(-human_fold$avg_log2FC), ]
#human_fold$avg_log2FC = rownames(human_fold)



id_human_all <- human_fold$avg_log2FC
names(id_human_all) <- rownames(human_fold)



MAST_Human_20 <- MAST_Human_20[order(-MAST_Human_20$avg_log2FC), ]
id_human_des <- MAST_Human_20$avg_log2FC
names(id_human_des) <- rownames(MAST_Human_20)



```
  
  
  
  
  
```{r} 

## fgsea with hallmark, GMT file from http://www.gsea-msigdb.org/gsea/msigdb/index.jsp
gmtfile <- "human_fRGC_pRGC/h.all.v7.4.symbols.gmt"
hallmark <- read.gmt(gmtfile)
hallmark$term <- gsub('HALLMARK_', '', hallmark$term)
hallmark.list <- hallmark %>% split(.$term) %>% lapply("[[", 2)


## Perform the fgsea analysis
set.seed(98177)
fgseaRes <- fgsea(
  pathways = hallmark.list,
  stats = id_human_all,
  minSize = 1,
  maxSize = 10000,
  nperm = 10000
)
sig_h  <- fgseaRes[fgseaRes$padj < 0.05, ]
sig_h <- sig_h[order(sig_h$NES, decreasing = T), ]

sig_h

```
  
  
  
## Plot the GESA pathway
  
```{r}


plotGseaTable(hallmark.list[sig_h$pathway], id_human_all,
              fgseaRes,
              gseaParam = 0.5)


#fwrite(sig_h, file ="hallmark_GESA.csv")



p1  = plotEnrichment(hallmark.list[["OXIDATIVE_PHOSPHORYLATION"]], id_human_all) + labs(title =
                                                                                          "OXIDATIVE_PHOSPHORYLATION") + theme_grey(base_size = 7)

p2 = plotEnrichment(hallmark.list[["GLYCOLYSIS"]], id_human_all) + labs(title =
                                                                          "GLYCOLYSIS") + theme_grey(base_size = 7)

p3 = plotEnrichment(hallmark.list[["HYPOXIA"]], id_human_all) + labs(title =
                                                                       "HYPOXIA") + theme_grey(base_size = 7)


multi.page <- ggarrange(p1,
                        p2,
                        p3,
                        nrow = 3,
                        ncol = 1)
multi.page



#graph2ppt(file = 'gsea.ppt',height = 4,width = 3)


```
  
  
  
  
```{r}

inquiry_1 = c(
  'TFAM','PPARA','PPARG','ERRFI1','SIRT1',
  'SIRT6','SIRT7','SIRT2','AMBRA1','BCL2',
  'MULL_p','MULL_n','LRRK2','FOXO4' ,'FOXO1' ,
  'FOXO3','ULK1' ,'ULK2','ATG2A' ,'ATG2B'
)

inquiry_2 = c(
  'ATG3' ,'ATG4A' ,'ATG4B','ATG5',
  'BECN1','ATG7','ATG9A','ATG9B',
  'ATG10','ATG12','ATG13','ATG14' ,
  'ATG16L1' ,'WIPI1','WIPI2_n', 'ATG101' ,
  'FIP200','KLF4','HDAC6','RAB35','RAB7A'
)


inquiry_3  = c(
  "HIF1A","SOD1","SOD2","SOD3",
  "PRDX4","PRDX6","TXN1","GSS",
  "GPX1","GPX2","GPX3","GPX4",
  "GPX5","GPX6","GPX7","GTSK1",
  "GSTP1","GSK3A","GSK3B"
)


inquiry_4 = c(
  "ISG15","LY6E","IFI6","ISG20","MX1",
 "IFIT2","IFIT1","CXCL10","CCL8")

inquiry_5= c(
  'PINK1','FUNDC1','OPA1','MFN1',
  'MFN2','OPTN','TAX1BP1','NBR1',
  'TOMM20','LAMP1_p','LAMP2','LAMP1_n',
  'NRF1','DRP2','HIF1A','BNIP3',
  'GABARAPL1','GABARAPL2'
)

total = c(inquiry_1, inquiry_2, inquiry_3, inquiry_4 , inquiry_5)

```



## tabulate the expression of inquiry genes
```{r}

Human_result = MAST_Human_20[row.names(MAST_Human_20) %in% total,]
Human_result <- Human_result[order(-Human_result$avg_log2FC),]

Human_result

```

  
  
  
  
  
  
  
  





























# Macaque dataï¼Œdata was downloaded from
https://singlecell.broadinstitute.org/single_cell/study/SCP212/molecular-specification-of-retinal-cell-types-underlying-central-and-peripheral-vision-in-primates#study-download
Macaque_fov_RGC_expression.txt
Macaque_per_RGC_expression.txt

```{r}
library(readr)
library(Seurat)
library(cowplot)
library("patchwork")

# import counting table
fRGC <- read_csv("~/Macaque_fov_RGC_expression.txt")
fRGC1 = fRGC[,-1]
genefrgc = fRGC[,1]
row.names(fRGC1) = fRGC$GENE 
fRGC = fRGC1


pRGC <- read_csv("~/Macaque_per_RGC_expression.txt")
pRGC1 = pRGC[, -1]   
row.names(pRGC1) = pRGC$GENE
pRGC = pRGC1


# Set up pRGC object
pRGC <- CreateSeuratObject(counts = pRGC, project = "pRGC")
Idents(object = pRGC) <- 'peripheral'



# Set up fRGC object
fRGC <- CreateSeuratObject(counts = fRGC, project = "fRGC")
Idents(object = fRGC)  = 'foveal'

#immune.anchors <- FindIntegrationAnchors(object.list = list(pRGC, fRGC), dims = 1:20)
#RGC <- IntegrateData(anchorset = immune.anchors, dims = 1:20)

# Merge the two Seurat objects
merged_seurat <- merge(pRGC, fRGC)
merged_seurat@active.ident <- factor(x = merged_seurat@active.ident, levels = c('peripheral', 'foveal'))



Macaque_RGC_2 = merged_seurat
   


```



```{r}
#gene differential analysis between foveal and peripheral cell 

MAST_Macaque_20 = FindMarkers(Macaque_RGC_2, ident.1 = "foveal", ident.2 = "peripheral", test.use = "MAST",verbose = FALSE)
MAST_Macaque_20$Gene = rownames(MAST_Macaque_20)

```





## Plot, Mitochondrial Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)


```{r}
mrp_List =  MAST_Macaque_20  %>%  filter( grepl('MRP', Gene))


EnhancedVolcano(mrp_List,
                lab = rownames(mrp_List) ,
                x = 'avg_log2FC',
                y = 'p_val_adj',
                xlim = c(-1, +1),
                ylim = c(0,300),
                title = "Mitochondrial Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)",
                FCcutoff = 0.24,
                labSize = 4,
                drawConnectors = TRUE,
                arrowheads = TRUE,
                colConnectors = 'black',
                legendPosition = 'top',
                legendLabSize = 10,
                legendIconSize = 5
)

# graph2ppt(file = 'monkey_Mitochondrial Ribosoma_Protein Genes.ppt',height = 7,width = 8.5)



```



## Plot, Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)

```{r}

library(EnhancedVolcano)
MAST_Macaque_20$NAME = rownames(MAST_Macaque_20)
rp_List =  MAST_Macaque_20 %>%  filter(grepl('^RP', NAME))
rp_List1 = MAST_Macaque_20  %>%  filter(grepl(paste(c('^RPL','^RPS'), collapse = "|"), NAME) )



d = EnhancedVolcano(rp_List1,
                    lab = rownames(rp_List1) ,
                    x = 'avg_log2FC',
                    y = 'p_val_adj',
                    xlim = c(-2, +2),
                    ylim = c(200,301),
                    title = "Ribosomal Protein Genes Up-regulated in fRGC (20% Threshsold)",
                    FCcutoff = 0.24,
                    labSize = 4,
                    drawConnectors = TRUE,
                    arrowheads = TRUE,
                    colConnectors = 'black',
                    legendPosition = 'top',
                    legendLabSize = 10,
                    legendIconSize = 5

)

d
# graph2ppt(file = 'monkey_Ribosomal Protein Genes.ppt',height = 7,width = 8.5)



```



```{r}
Macaque_fold = FoldChange(Macaque_RGC_2 , 
                         ident.1 = "foveal",
                         ident.2 = "peripheral")
Macaque_fold <- monkey_fold[order(-Macaque_fold$avg_log2FC),]
Macaque_fold$gene = rownames(Macaque_fold)

id_Macaque_all <- Macaque_fold$avg_log2FC
names(id_Macaque_all) <- rownames(Macaque_fold)




## fgsea for hallmark 
setwd("~/")


gmtfile <- "./msigdb.v2022.1.Mm.symbols.gmt"

hallmark <- read.gmt(gmtfile)
hallmark$term <- gsub('HALLMARK_','',hallmark$term)
hallmark.list <- hallmark %>% split(.$term) %>% lapply( "[[", 2)




## Perform the fgsea analysis
set.seed(98177)
fgseaRes <-fgseaMultilevel(pathways = hallmark.list, 
                           stats = id_monkey_all,
                           minSize=1,
                           maxSize=10000)
sig_Macaque <- fgseaRes[fgseaRes$padj<0.05,]
sig_Macaque <- sig_monkey[order(sig_Macaque$NES,decreasing = T),]


## plot
plotGseaTable(hallmark.list[sig$pathway],id, fgseaRes,gseaParam = 0.5)


```




## tabulate the expression of inquiry genes
```{r}

Macaque_result = MAST_Macaque_20[row.names(MAST_Macaque_20) %in% total,]
Macaque_result <- Macaque_result[order(-Human_result$avg_log2FC),]

Macaque_result

```



# violin plots for comparison of 'HIF1A','OPTN','SOD1','IFI6' in human / macaque data

```{r}
# PPT for violin plot

p = VlnPlot(
  Human_Object,
  features = c('HIF1A','OPTN','SOD1','IFI6'),
  assay = "RNA",
  pt.size = 0,
  ncol = 4
) 
p

p2 = VlnPlot(
  Macaque_RGC_2,
  features = c('HIF1A','OPTN','SOD1','IFI6'),
  assay = "RNA",
  pt.size = 0,
  ncol = 4
) 
p2 


library(ggpubr)
multi.page <- ggarrange(p, p2,
                        nrow = 2, 
                        ncol = 1) + ggtitle("Human(top) and Monkey(down)") + theme(plot.title = element_text(lineheight=.8, face="bold"))


multi.page


#graph2ppt(file = 'volin_Genes.ppt',height = 7,width = 8.5)

```








